components/identity-mgt/org.wso2.carbon.identity.mgt/src/main/java/org/wso2/carbon/identity/mgt/IdentityMgtEventListener.java

```diff
@@ -22,6 +22,7 @@
 import org.apache.commons.lang.StringUtils;
 import org.apache.commons.logging.Log;
 import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.context.CarbonContext;
```

Comment:
This the first place you introduce CarbonContext in the listener code. Other places its picks the tenant from the user store manager instance. Better to follow the same approach here as well. Ref following,

`String tenantDomain = IdentityTenantUtil.getTenantDomain(userStoreManager.getTenantDomain());`